modules/artifact/src/main/java/org/mule/runtime/module/artifact/internal/descriptor/SupportedJvmArtifactDescriptorValidator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.internal.descriptor;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.api.util.MuleSystemProperties.SYSTEM_PROPERTY_PREFIX;
+
+import static java.lang.String.format;
+import static java.lang.System.getProperty;
+
+import static org.apache.commons.lang3.SystemUtils.JAVA_SPECIFICATION_VERSION;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.module.artifact.api.descriptor.ApplicationDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptorValidator;
+import org.mule.runtime.module.artifact.api.descriptor.DeployableArtifactDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.DomainDescriptor;
+
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Validator for {@link DeployableArtifactDescriptor#getSupportedJavaVersions()}. Checks the currently running Java version
+ * against the values provided by the descriptor.
+ * 
+ * @since 4.6
+ */
+public class SupportedJvmArtifactDescriptorValidator implements ArtifactDescriptorValidator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SupportedJvmArtifactDescriptorValidator.class);
+
+  /**
+   * System property to set the enforcement policy. Defined here as a decision was made not to expose it as an API yet. For now,
+   * it will be for internal use only.
+   *
+   * @since 4.5.0
+   */
+  static final String DEPLOYABLE_ARTIFACT_JVM_ENFORCEMENT_PROPERTY = SYSTEM_PROPERTY_PREFIX + "jvm.version.extension.enforcement";
```

Comment:
Should this property be the same as the one for Extensions? if that is the case, shouldn't we change the name to something more generic that does not include the word `extension` in it.